hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -269,9 +277,13 @@ protected Option<BufferedRecord<T>> doProcessNextDataRecord(BufferedRecord<T> ne
       } else {
         switch (recordMergeMode) {
           case COMMIT_TIME_ORDERING:
+            updatePartiallyIfNeeded(
```

Comment:
L 247 - 276 is where we already do partial merging. 
We should move all these logic in there. 
or have a high level switch case for merge mode, and for each merge mode , we handle both full merge and partial merge based on configs.

Reply:
I see the TODO "TODO(HUDI-7843): decouple the merging logic from the merger",

is it possible we introduce a new merge mode for partial merge and handle all the partial merging use cases there? so that to make the other branches clean(no redundan ramifications in row-level).